src/backend/distributed/executor/multi_utility.c

```diff
@@ -1310,6 +1517,58 @@ ExecuteDistributedDDLCommand(Oid relationId, const char *ddlCommandString,
 
 
 /*
+ * Inter table DDL command is a DDL command which contains two tables in it. For example
+ * ALTER TABLE table1 ADD FOREIGN KEY key_name(column_name) REFERENCES table2(column_name)
+ * query refers two tables. ExecuteDistributedInterTableDDLCommand applies a given inter
+ * table DDL command in a distributed transaction. To be able to execute such query, given
+ * two tables should be co-located. If they are not co-located, this function gives an
+ * error message.
+ *
+ * If the multi shard commit protocol is in its default value of '1pc', then a notice
+ * message indicating that '2pc' might be used for extra safety. In the commit protocol,
+ * a BEGIN is sent after connection to each shard placement and COMMIT/ROLLBACK is handled
+ * by CompleteShardPlacementTransactions function.
+ */
+static void
+ExecuteDistributedInterTableDDLCommand(Oid leftTableId, Oid rightTableId,
+									   const char *ddlCommandString)
+{
+	bool executionOK = false;
+
+	if (XactModificationLevel == XACT_MODIFICATION_DATA)
+	{
+		ereport(ERROR, (errcode(ERRCODE_ACTIVE_SQL_TRANSACTION),
+						errmsg("distributed DDL commands must not appear within "
+							   "transaction blocks containing data modifications")));
+	}
+
+	if (!TablesColocated(leftTableId, rightTableId))
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("cannot execute a DDL command which refers two tables"),
+						errdetail("DDL commands refering two tables can only be executed "
+								  "on co-located tables."),
+						errhint("If these tables are physically co-located try mark them"
+								"as co-located with mark_tables_colocated() function.")));
```

Comment:
We currently don't have `mark_tables_colocated()` and probably won't have on 6.0.

We may not share a hint ?